syft/pkg/cataloger/common/cpe/java.go

```diff
@@ -184,6 +184,113 @@ func GroupIDsFromJavaPackage(p pkg.Package) (groupIDs []string) {
 	return GroupIDsFromJavaMetadata(p.Name, metadata)
 }
 
+// GroupIDFromJavaPackage returns the authoritative group ID for a Java package.
+// The order of precedence is:
+// 1. The group ID from the POM properties
+// 2. The group ID from the POM project
+// 3. The group ID from a select map of known group IDs
+// 3. The group ID from the Java manifest
+func GroupIDFromJavaMetadata(pkgName string, metadata pkg.JavaMetadata) (groupID string) {
```

Comment:
this function is missing new tests.

Reply:
Added some base cases in [1e4f54a](https://github.com/anchore/syft/pull/2033/commits/1e4f54a99f5a3f9e89c4e2aa8318dfa7afb0a2b1)